### PR TITLE
GLTFLoader: Parsing provided data only if is not in JSON format.

### PR DESCRIPTION
**Pull request by antonio-gomez**
Fix for #12470

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -74,14 +74,22 @@ THREE.GLTFLoader = ( function () {
...
-			}
+					var json = JSON.parse( data );
+
+				} catch ( e ) {
 
-			var json = JSON.parse( content );
+					content = convertUint8ArrayToString( new Uint8Array( data ) );
+					var json = JSON.parse( content );
+
+				}
```
[the remark this answers is not in the document]
Comment (antonio-gomez) [reply examples/js/loaders/GLTFLoader.js]: Cool, let me make the fix and I can update the pull request.

**Review by antonio-gomez — COMMENTED** [review]
Hey @donmccurdy, I applied the fix to the Parse method and updated the PR. Thanks!

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -70,18 +70,28 @@ THREE.GLTFLoader = ( function () {
...
+				} else {
+
+					content = convertUint8ArrayToString( new Uint8Array( data ) );
+
+				}
 
 			}
 
-			var json = JSON.parse( content );
+				var json = JSON.parse( content );
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: Wrong indentation here

Comment (antonio-gomez) [reply examples/js/loaders/GLTFLoader.js]: Pushed a fix for this @takahirox. Thanks!

File: examples/js/loaders/GLTFLoader.js [hunk 3]
```diff
@@ -70,14 +70,24 @@ THREE.GLTFLoader = ( function () {
 
 			var magic = convertUint8ArrayToString( new Uint8Array( data, 0, 4 ) );
 
-			if ( magic === BINARY_EXTENSION_HEADER_MAGIC ) {
+			if ( typeof data === 'string' ) {
 
-				extensions[ EXTENSIONS.KHR_BINARY_GLTF ] = new GLTFBinaryExtension( data );
-				content = extensions[ EXTENSIONS.KHR_BINARY_GLTF ].content;
+					content = data;
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: Wrong indentation here, too

**Comment by donmccurdy** [comment]
Indent is fixed, this looks good to me. Thanks!

**Review by donmccurdy — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks everyone!

**Pull request merged**